test/tests/functional/pbs_reservations.py

```diff
@@ -85,56 +86,324 @@ def submit_asap_reservation(self, user, jid):
+        self.server.manager(MGR_CMD_SET, NODE, a, id=resv_node)
+
+        a = {'reserve_substate': 10}
+        a.update(resv_state)
+        self.server.expect(RESV, a, id=rid)
+
+        self.scheduler.log_match(rid + ';Reservation is in degraded mode',
+                                 starttime=now, interval=1)
+
+        self.server.expect(RESV, a, id=rid)
+
+        self.server.expect(RESV, {'resv_nodes':
+                                  (MATCH_RE, re.escape(resv_node))}, id=rid)
+
+        if rrule and run:
+            a = {'reserve_state': (MATCH_RE, 'RESV_DEGRADED|10'),
+                 'reserve_substate': 10}
+            t = end - int(time.time())
+            self.logger.info('Sleeping until reservation ends')
+            self.server.expect(RESV, a, id=rid, offset=t)
+
+        self.server.manager(MGR_CMD_SET, NODE, {'state': (DECR, 'offline')},
+                            id=resv_node)
+        # If run and rrule are true, we waited until the occurrence
+        # finished and the reservation is no longer running otherwise
+        # the reservation is still running.
+        if run:
+            if rrule:
+                resv_state = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2'),
+                              'reserve_substate': 2}
+            else:
+                resv_state = {'reserve_state': (MATCH_RE, 'RESV_RUNNING|5'),
+                              'reserve_substate': 5}
+        else:
+            resv_state = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2'),
+                          'reserve_substate': 2}
+
+        self.server.expect(RESV, resv_state, id=rid)
+
+    def degraded_resv_node_down_back(self, start, end, rrule=None, run=False):
+        """
+        Verify that when a node goes offline and comes back, the reservation
+        goes to the degraded state and then back to the correct state
+        """
+
+    def test_degraded_standing_reservations(self):
+        """
+        Verify that degraded standing reservations are reconfirmed
+        on other nodes
+        """
+        t = int(time.time())
+        self.degraded_resv_reconfirm(start=t + 25, end=t + 625,
+                                     rrule='freq=HOURLY;count=5')
+
+    def test_degraded_advanced_reservations(self):
+        """
+        Verify that degraded advanced reservations are reconfirmed
+        on other nodes
+        """
+        t = int(time.time())
+        self.degraded_resv_reconfirm(start=t + 25, end=t + 625)
+
+    def test_degraded_standing_running_reservations(self):
+        """
+        Verify that degraded standing reservations are reconfirmed
+        on other nodes
+        """
+        t = int(time.time())
+        self.degraded_resv_reconfirm(start=t + 25, end=t + 625,
+                                     rrule='freq=HOURLY;count=5', run=True)
+
+    def test_degraded_advanced_running_reservations(self):
+        """
+        Verify that degraded advanced reservations are reconfirmed
```

Comment:
I think the description here is wrong, such a reservation will not be reconfirmed.